consensus/service/src/byzantine_ledger/worker.rs

```diff
@@ -795,6 +805,70 @@ impl<
         counters::CUR_SLOT_NOMINATION_ROUND.set(slot_metrics.cur_nomination_round as i64);
         counters::CUR_SLOT_BALLOT_COUNTER.set(slot_metrics.bN as i64);
     }
+
+    fn form_block_from_externalized_values(
+        &self,
+        externalized_values: Vec<ConsensusValue>,
+    ) -> BlockData {
+        let num_blocks = self
+            .ledger
+            .num_blocks()
+            .expect("Ledger must contain a block.");
+        let parent_block = self
+            .ledger
+            .get_block(num_blocks - 1)
+            .expect("Ledger must contain a block.");
```

Comment:
Optional/Equivalent: Change this message to make it easier to differentiate which call failed.

```suggestion
            .expect("Failed to get last block.");
```